### ISSUE TITLE
Removed button colors on v144.0.0

### ISSUE THREAD

**Issue by seahindeniz** [brainly/style-guide#1584]
Hello guys and @burakukula !

_**TL;DR:**_
Is it fully necessary to remove the peach color from the style guide and could you make it available again?

---

>I understand that the colors that you've removed in the #1581 may/are not used in the company based(internal) applications.
And I also know that the style-guide is not ment to be only for the company based projects, otherwise it wouldn't be available on the Github as a public repository; am I right?
Other people can able to use the style-guide in everywhere.
Since I'm using those colors- especially the Peach colored button in my project(like a lot) and your changes also affecting it. 

I have a project and it's for a small group of users of Brainly and it's called Brainly Tools. Espacially for the moderators, admins and the content managers. It's a browser extension and it gives us a bit more better user experience and it's allow us to make proper actions faster than usual. This makes users happy enough :).

My project uses the style-guide.css file that loads under the head tag in new brand pages. Like, feed/home page or the question view page.
But my project is also uses the style-guide.css file in the legacy pages like the user profile page or the user contents page by injecting the css file in the head tag.

Whenever you change something from the style guide, it makes me worried as you can imagine :sweat_smile:
The buttons that I've used in the new brand pages is based on the _untouched_ style-guide.css file that comes from the head tag. The legacy pages uses the injected css file.

So I would like to ask; is it fully necessary to remove the peach color from the style guide and could you make it available again?

Please see the example usage in the screenshots below

**Before**:
![image](https://user-images.githubusercontent.com/5789670/58881788-88a93700-86e3-11e9-81ee-41c7461c04ee.png)

**After**:
![image](https://user-images.githubusercontent.com/5789670/58881859-bbebc600-86e3-11e9-83cb-2ef467202b16.png)

**Comment by burakukula** [comment]
hey @seahindeniz! 🙂Nice to hear that you use our style-guide for your widget! Awesome! 🚀
I guess, in general, if you don't want to be affected by our changes right away, you could also use versioning in your project as we do in the main application or encapsulate your widget somehow.
Actually, the changes that you mention (on the second screenshot) are not on production yet, we are still testing them. The current version of style-guide on production is still on `https://styleguide.brainly.com/143.9.1/style-guide.css`. Do you have versioning on your side? 
Regarding the changes in the buttons - colors, and styling, decisions are done by our design team, so we don't have much impact on that. But the good news is that in the new system of buttons - that gonna be introduced hopefully soon, you would have an option for the `warning` button (that have peach color but rounded). There will be a separate CSS class for that type of button, so you would need to adjust your code a little bit. 
I'll let you know as soon, as new buttons (the rounded once, that are still in the phase of tests) are released to production.

**Comment by seahindeniz** [comment]
Thanks for reply @burakukula 

Yes, I'm using versioning in my project and to be honest, my project fetches the **package.json** file of this repository and uses the current version number to fetch style-guide.css file with this dynamic way
```js
// shortly gulpfile.babel.js
import syncReq from "sync-request";
import downloadFileSync from "download-file-sync";

const styleGuidePJBody = syncReq("GET", "https://raw.githubusercontent.com/brainly/style-guide/master/package.json");

const styleGuidePJ = JSON.parse(styleGuidePJBody.getBody('utf8'));

const styleGuideContent = downloadFileSync(`https://styleguide.brainly.com/${styleGuidePJ.version}/style-guide.css`);
```

And this is for making the style guide components available on the legacy UI by storing the style-guide.css file as a local file in the extension storage and injecting it to DOM before page load.
>style-guide.css file is locally stored because injecting the file with this way decreases the loading time


For the newer UI, my extension uses the components of the style-guide.css that loaded in DOM. That means, extension doesn't have a control of <style \/> tag of the style-guide.css file in the <head \/> tag added by Brainly.

I see several options to make things possible like: by blocking the brainly's latest style-guide.css file before page load and injecting the locally stored version of style-guide.css file into <head \/> tag _or_ by adding some custom colored styles in a css file to making the coloring possible for the buttons _or_ etc..

_Blocking the style-guide file is not a good solution for me to apply since latest changes on the homepage may have affected by this way._

Anyways, I'm waiting for the news :heavy_heart_exclamation:

**Comment by burakukula** [comment]
@seahindeniz, sure I get it, the one issue that I see here, is that, as in that case, version of the style guide is different on production, that the last release of the library. you are running version`144.0.0` in your widget, so the last release, but we still did not bump it and we are on the `143.9.1`. This type of situations almost never happen, so we should not run into this problem often though.
Consider also, that style guide is driven by our design, and there always can be some breaking changes, like in this case, components get deprecated and so on, so it might be good for you to have more control over versioning, so you can see what actually changed on the style-guide side. 
Our buttons situation will be solved within the next 2-3 weeks. I'll keep you posted. 🙂

**Comment by burakukula** [comment]
Hey @seahindeniz! small update from our side, sorry it took so long. 
In the current version of SG `145.1.0` there is a new button component available - https://styleguide.brainly.com/145.1.0/docs/components.html#buttons. It has new namespaces (`sg-button` with its modifiers) together with new CSS, so it requires some updates on your side I guess, but it might help you to solve some problems with styling, that occurred after our last big update - e.g `sg-button sg-button--destructive` which have our `peach` color. There is a very simple readme for buttons available https://github.com/brainly/style-guide/blob/master/src/components/buttons/README.md - work on documentation is still in progress, I hope it will be extended soon, but maybe you find it useful in the current state anyway.
If you need any help from our side, please let us know.
Cheers!

**Comment by seahindeniz** [comment]
Thanks for letting me know and no worries about the delay, @burakukula 

Here, thanks to your changes I started using the new button component..
<img src="https://user-images.githubusercontent.com/5789670/61010585-817ae600-a37f-11e9-8559-5eada211b35c.png" width="50%" height="50%">

Although but I have a question; do you have plans for the `sg-button-primary` and `sg-button-secondary`.
Like a date of removal for these components from SG or is it going to be removed at all?..

**Comment by burakukula** [comment]
hey @seahindeniz! Great to hear that new buttons are used by your project already.
We are currently working to replace old components with the new ones in the main. Later on, we will remove `sg-button-primary` and `sg-button-secondary`. 
They will be available for some time, just in case we miss something, but I think the best for you would be to consider them as deprecated from now on.
